torchvision/prototype/transforms/_transform.py

```diff
@@ -32,6 +35,8 @@ def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
     def forward(self, *inputs: Any) -> Any:
         sample = inputs if len(inputs) > 1 else inputs[0]
 
+        self._check(sample)
```

Comment:
Why aren't we flattening before this call? Also in this approach we can avoid the multiple flattening calls within `has_any`, `has_all` and the rest of the utility methods. Do you plan to do this on a follow up?